src/characters/creatures/FireFly.ts

```diff
@@ -0,0 +1,38 @@
+import * as Phaser from 'phaser';
+
+export default class FireFly extends Phaser.Physics.Arcade.Sprite {
+    private velX: number
+    private velY: number
+
+    constructor(scene: Phaser.Scene, x: number, y: number, texture: string, frame?: string | number) {
+      super(scene, x, y, texture, frame);
+
+      this.setRandomDirection();
+      scene.physics.world.enable(this);
+      this.setCollideWorldBounds(true);
+    }
+
+    setRandomDirection() {
```

Comment:
Seems to be redundant function - just assign velocity in constructor. Also, please, use full words for variables names, like this.velocityX instead of this.velX